modules/gapi/test/common/gapi_render_tests.hpp

```diff
@@ -12,6 +12,36 @@
 #include "api/render_priv.hpp"
 #include "api/render_ocv.hpp"
 
+namespace opencv_test {
+
+using Prims = cv::gapi::wip::draw::Prims;
+using Prim  = cv::gapi::wip::draw::Prim;
+
+/** GTest parameter for Prims */
+struct PrimsTestParam
+{
+    enum Value {
+        RECTS,
+        CIRCLES,
+        LINES,
+        MOSAICS,
+        IMAGES,
+        POLYGONS,
+        TEXTS
+    } val_;
+
+    PrimsTestParam(Value val) : val_(val) {}
+    void PrintTo(std::ostream* os) const
+    {
+        static const char* svals[] = { "RECTS", "CIRCLES", "LINES", "MOSAICS", "IMAGES", "POLYGONS", "TEXTS" };
+        if (val_ >= 0 && val_ < (sizeof(svals) / sizeof(svals[0])))
+            *os << svals[val_];
+        else
+            *os << "UNKNOWN(" << (int)val_ << ")";
```

Comment:
I'd understand you autogenerate labels from enum and then use it as a lookup, but _this_ way it is not better (actually, is worse) than just a plain `switch()`. I'd recommend to write it with a `switch()`.

Reply:
This is a comment for @TolyaTalamanov